src/cpp/session/modules/NotebookData.R

```diff
@@ -73,6 +73,10 @@
 
     x <- .rs.toDataFrame(head(x, max.print), "x", flatten = FALSE, force = TRUE)
 
+    if (NCOL(x) > max.print) {
```

Comment:
I agree as well; I don't think `max.print` is used to cap columns anywhere else.